External/FEXCore/Source/Interface/Core/ObjectCache/CodeObjectSerializationConfig.h

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <cstdint>
+
+namespace FEXCore::CodeSerialize {
+  // If any of the config options mismatch on load then the cache won't be used
+  // Any of these will result in codegen changes
+  struct CodeObjectSerializationConfig {
```

Comment:
Would be great to have this harmonized with the AOTIR format.

Reply:
Could probably share some of the config but not all of it. Once this config pulls in host features then there is a mismatch between what config effects IR versus which config options effects assembly.

Reply:
Yeah, we could split into two parts one for "IR-abi" and one for "object-abi".

Would also be nice if we used the same postfix notations in the cache files in both.

There's a further slight complication there with #1558, `MMAN` and `MTRACK` smc modes don't change the ABI at all.

Reply:
(But are different `SMCChecks` values)

Reply:
That's fine, doesn't need to be flawless.